cli/docker_validator.py

```diff
@@ -0,0 +1,205 @@
+    try:
+        run_command_with_output("pull container image", f"{docker} pull {container_image}")
+        if not run_container(docker, container_name):
+            raise RuntimeError(f"Container ({container_name}) could not be started")
+    except subprocess.CalledProcessError as ex:
+        logging.error(ex)
+        raise RuntimeError(f"Cannot pull container ({container_image}) image")
+
+
+def verify_docker_env(docker, container_name):
+    verify_docker_is_installed(docker)
+
+    if not verify_container_is_running(docker, container_name):
+        if verify_container_is_stopped(docker, container_name):
+            try_restarting_container(docker, container_name)
+        else:
+            if verify_image_available_locally(docker, container_image):
+                run_container(docker, container_name)
+            else:
+                download_container_image(docker, container_name)
+
+def copy_files_to_container(docker, container_name, mapping_file):
+    run_command_with_output("Create directory structure for copying vcf metadata file into container",
+                            f"{docker} exec {container_name} mkdir -p {container_validation_dir}/{mapping_file.parent}")
+    run_command_with_output("Copy vcf metadata file to container",
+                            f"{docker} cp {mapping_file} {container_name}:{container_validation_dir}/{mapping_file}")
+
+    with open(mapping_file) as open_file:
+        reader = csv.DictReader(open_file, delimiter=',')
+        for row in reader:
+            run_command_with_output("Create directory structure to copy vcf files into container",
+                                    f"{docker} exec {container_name} "
+                                    f"mkdir -p {container_validation_dir}/{Path(row['vcf']).parent}")
+            run_command_with_output("Copy vcf file to container",
+                                    f"{docker} cp {row['vcf']} {container_name}:{container_validation_dir}/{row['vcf']}")
+            run_command_with_output("Create directory structure to copy fasta files into container",
+                                    f"{docker} exec {container_name} "
+                                    f"mkdir -p {container_validation_dir}/{Path(row['fasta']).parent}")
+            run_command_with_output("Copy fasta file to container",
+                                    f"{docker} cp {row['fasta']} {container_name}:{container_validation_dir}/{row['fasta']}")
+
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser(description='Run pre submission validation checks on vcf files', add_help=False)
+    parser.add_argument("--docker_path", help="full path to the docker installation, "
+                                              "not required if docker is available on path", required=False)
+    parser.add_argument("--container_name", help="name of the docker container", required=False)
+    parser.add_argument("--vcf_files_mapping",
```

Comment:
This seems like extra overhead for submissions with a single VCF. Can we not ask submitters to put their VCF and metadata file in the same folder like we do with private FTP?

Reply:
The idea was to give flexibility to the user to keep the files in different directories. 
If this feels like an overkill here, we can discuss on mounting the volume